packages/block-editor/src/store/actions.js

```diff
@@ -1461,6 +1461,22 @@ export const __unstableSetEditorMode =
 			);
 		} else if ( mode === 'zoom-out' ) {
 			speak( __( 'You are currently in zoom-out mode.' ) );
+			const { sectionRootClientId } = select.getSettings();
+			const sectionsClientIds =
+				select.getBlockOrder( sectionRootClientId );
+			registry.batch( () => {
+				dispatch.setBlockEditingMode(
+					'' /* rootClientId */,
+					'contentOnly'
+				);
+				dispatch.setBlockEditingMode(
+					sectionRootClientId,
+					'contentOnly'
+				);
+				sectionsClientIds.forEach( ( clientId ) =>
+					dispatch.setBlockEditingMode( clientId, 'default' )
+				);
```

Comment:
This logic should ideally move to the `getBlockEditingMode` selector.

Reply:
Fixed in #59249